### PR TITLE
Adding tests part2

### PR DESCRIPTION
**Pull request by RaEzhov**
Added tests for math_function. Also made some bugfixes.

### CODE REVIEW

File: source/math/calculus_parser.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def _find_pattern(self, query: str, pattern_dict: dict, try_predict: bool) -> bo
 
                     # Check if listed variables are correct
                     for var in symbols:
-                        if not str(var).isalpha():
+                        if not str(var).isalpha() or not str(var).isascii():
                             raise ParseError(f"Variables can only contain letters\nIncorrect variable: '{var}'")
```
Comment (Korostast) [inline source/math/calculus_parser.py:101]: We allow only latin variables? Then we should tell user about it
```suggestion
                        if not str(var).isalpha() or not str(var).isascii():
                            raise ParseError(f"Variables can only contain latin letters\nIncorrect variable: '{var}'")
```

File: source/math/math_function.py [hunk 2]
```diff
@@ -253,15 +253,15 @@ def slant_asymptotes(self, symbol: sy.Symbol) -> set:
...
-            if b.is_finite:
+            if b.is_number:
                 ans.add(k * symbol + b)
 
         k = sy.limit(self.simplified_expr / symbol, symbol, -sy.oo)
-        if k.is_finite:
+        if k.is_number:
             b = sy.limit(self.simplified_expr - k * symbol, symbol, -sy.oo)
-            if b.is_finite:
+            if b.is_number:
```
Comment (Korostast) [inline source/math/math_function.py:264]: We must add "**and k.is_finite**". Github doesn't allow to make a suggestion on deleted lines, so it cannot be done automatically

**Review by Korostast — APPROVED** [review]
I have done small corrections, but everything else is okay

File: source/math/math_function.py [hunk 3]
```diff
@@ -253,15 +253,15 @@ def slant_asymptotes(self, symbol: sy.Symbol) -> set:
         # TODO periodic function
 
         k = sy.limit(self.simplified_expr / symbol, symbol, sy.oo)
-        if k.is_finite:
+        if k.is_number:
```
Comment (RaEzhov) [inline source/math/math_function.py:256]:
```suggestion
        if k.is_number and k.is_finite:
```

File: source/math/math_function.py [hunk 4]
```diff
@@ -253,15 +253,15 @@ def slant_asymptotes(self, symbol: sy.Symbol) -> set:
...
             b = sy.limit(self.simplified_expr - k * symbol, symbol, sy.oo)
-            if b.is_finite:
+            if b.is_number:
                 ans.add(k * symbol + b)
 
         k = sy.limit(self.simplified_expr / symbol, symbol, -sy.oo)
-        if k.is_finite:
+        if k.is_number:
```
Comment (RaEzhov) [inline source/math/math_function.py:262]:
```suggestion
        if k.is_number and k.is_finite:
```

**Pull request merged**